### PR TITLE
doc: relnotes: 1.14 settings, flash_map, nRF5 flash driver and subsys documentation

### PR DESCRIPTION
**Pull request by nvlsianpu**

### CODE REVIEW

File: doc/releases/release-notes-1.14.rst [hunk 1]
```diff
@@ -159,6 +161,15 @@ Build and Infrastructure
 Libraries / Subsystems
 ***********************
 
+* flash_map:
+  - API extension
+  - Automatically generate of the list of flash areas
```
Comment (ioannisg) [inline doc/releases/release-notes-1.14.rst]: Automatic generation of

or

Automatically generate

File: doc/releases/release-notes-1.14.rst [hunk 2]
```diff
@@ -159,6 +161,15 @@ Build and Infrastructure
...
+
+* settings:
+  - enable Looging instead of ASSERTs
+  - always uses the storage partition for FCB
```
Comment (ioannisg) [inline doc/releases/release-notes-1.14.rst]: use

**Review by ioannisg — CHANGES_REQUESTED** [review]
style fixes

File: doc/releases/release-notes-1.14.rst [hunk 3]
```diff
@@ -159,6 +161,15 @@ Build and Infrastructure
 Libraries / Subsystems
 ***********************
 
+* flash_map:
+  - API extension
+  - Automatically generate of the list of flash areas
+
+* settings:
+  - enable Looging instead of ASSERTs
```
Comment (ioannisg) [inline doc/releases/release-notes-1.14.rst]: logging

**Comment by nvlsianpu** [comment]
^^ update

**Review by ioannisg — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by carlescufi** [comment]
Too fast for me :) Thanks @nvlsianpu!

**Pull request merged**